certora/spec/CVL2/timelockAuthorizerSasha.spec

```diff
@@ -0,0 +1,296 @@
+                => (isExecuted1Before == isExecuted1After 
+                    && isExecuted2Before == isExecuted2After 
+                    && isCancelled1Before == isCancelled1After)
+                    && !isCancelled2Before;
+}
+
+
+// STATUS - verified
+// Flag was changed by correct function. ("only one flag can be changed at a time" was checked above)
+rule onlyExecuteAndCancelCanChangeTheirFlags(env e, method f) {
+    uint256 actionIndex1; uint256 actionIndex2;
+    bool isExecuted1Before = getSchedExeExecuted(actionIndex1);
+    bool isCancelled1Before = getSchedExeCancelled(actionIndex1);
+
+    require actionIndex1 != actionIndex2;
+    require limitArrayLength();
+
+    calldataarg args;
+    f(e, args);
+
+    bool isExecuted1After = getSchedExeExecuted(actionIndex1);
+    bool isCancelled1After = getSchedExeCancelled(actionIndex1);
+
+    assert (isExecuted1Before != isExecuted1After)
+            => f.selector == sig:execute(uint256).selector;
+    assert (isCancelled1Before != isCancelled1After)
+            => f.selector == sig:cancel(uint256).selector;
+}
+
+
+// STATUS - verified
+// if canExecute returns true / false, execute finishes successfully / reverts.
+rule canExecuteAndExecuteUnion(env e, method f) {
+    uint256 scheduledExecutionId;
+
+    bool canExe = canExecute@withrevert(e, scheduledExecutionId);
+    bool canReverted = lastReverted;
+
+    execute@withrevert(e, scheduledExecutionId);
+    bool exeReveted = lastReverted;
+
+    assert (!canExe || canReverted) => exeReveted;
+}
+
+
+// STATUS - verified
+// If it’s one of the states (executed/cancelled) it should remain in forever.
+rule executedCanceledForever(env e, method f) {
+    uint256 actionIndex1;
+
+    require actionIndex1 < getSchedExeLength();
+    require limitArrayLength(); 
+
+    bool isExecutedBefore = getSchedExeExecuted(actionIndex1);
+    bool isCancelledBefore = getSchedExeCancelled(actionIndex1);
+
+    calldataarg args;
+    f(e, args);
+
+    bool isExecutedAfter = getSchedExeExecuted(actionIndex1);
+    bool isCancelledAfter = getSchedExeCancelled(actionIndex1);
+
+    assert isExecutedBefore && !isCancelledBefore => isExecutedAfter && !isCancelledAfter;
+    assert !isExecutedBefore && isCancelledBefore => !isExecutedAfter && isCancelledAfter;
+}
+
+
+// STATUS - verified
+// account with GLOBAL_CANCELER_SCHEDULED_EXECUTION_ID() can cancel any scheduled execution.
+rule almightyGlobal(env e, uint256 scheduledExecutionId) {
+    bool isAlmighty = _isCanceler(GLOBAL_CANCELER_SCHEDULED_EXECUTION_ID(), e.msg.sender);
+    require !_isCanceler(scheduledExecutionId, e.msg.sender);   // eliminate scenarious when a sender is a canceler for a specific scheduled execution
+    require !isRoot(e.msg.sender);   // eliminate scenarious when a sender is a root
+
+    cancel@withrevert(e, scheduledExecutionId);
+    bool isReverted = lastReverted;
+
+    assert !isReverted => isAlmighty, "Remember, with great power comes great responsibility.";
```

Comment:
In my last job I would be certainly told, that I should avoid funny comments, that are unnecessary but I feel you. ;)

Reply:
It's just an error message. It's in my snippet. Most of the time, I remove it, but sometimes I forget.